### PR TITLE
Add bit indicating whether orig packet used VXLAN or VXLAN-GPE encap

### PR DESCRIPTION
**Pull request by mickeyspiegel**
Include guidelines suggesting that the sink should ignore this bit in
mixed v1.0 / v1.1 environments.

### CODE REVIEW

File: telemetry/specs/INT.mdk [hunk 1]
```diff
@@ -711,6 +711,26 @@ INT shim header for VXLAN GPE encapsulation:
 * Type: This field indicates the type of INT Header following the shim header.
 Two Type values are used: one for the hop-by-hop header type and the other for
 the destination header type (See Section [#sec-int-header-types]).
+* G: Indicates whether the original packet (before insertion of INT headers
+and metadata) used a VXLAN or VXLAN GPE encapsulation.
+  - 0: Original packet used VXLAN encapsulation.
+  - 1: Original packet used VXLAN GPE encapsulation.
```
Comment (mhira1) [inline telemetry/specs/INT.mdk]: Wondering whether we should flip this and use 0 to indicate the original packet was VXLAN GPE, i.e. no conversion is required and 1 to indicate that original packet was VXLAN, i.e. yes conversion is required at INT sink. Thus INT sink does VXLAN GPE to VXLAN conversion if this bit is set to 1.

**Review by mhira1 — COMMENTED** [review]
Looks good, minor comment inline.

**Comment by jklr** [comment]
@mickeyspiegel
This is a reminder of the AR from the May meeting: All text around 1.0/1.x interoperability needs to be modified as this is now going into version 2.0. Mickey to revise, Mukesh to review and merge.

File: telemetry/specs/INT.mdk [hunk 2]
```diff
@@ -711,6 +711,26 @@ INT shim header for VXLAN GPE encapsulation:
...
+    value of this bit and rely on local configuration to determine whether to
+    convert the VXLAN GPE encapsulation back to a VXLAN encapsulation.
+    Typically in such a mixed or v1.0 environment, INT would only be inserted
+    into one type of encapsulation but not the other, e.g. VXLAN but not
+    VXLAN GPE since the INT sink always converts VXLAN GPE packets with INT
+    back to the VXLAN encapsulation.
+  - In an all v1.1 environment, both packets received with VXLAN encapsulation
+    and VXLAN GPE encapsulation could have INT inserted, with the INT sink
+    relying on the value of this bit to determine whether to carry out the
+    conversion from VXLAN GPE back to VXLAN.
```
Comment (mickeyspiegel) [inline telemetry/specs/INT.mdk]: Since this is now aimed at the INT v2.0 spec, the two bullet points about v1.0, v1.1 and interop can just be removed.

Comment (jklr) [reply telemetry/specs/INT.mdk]: +1

File: telemetry/specs/INT.mdk (hunk 1 above)
Comment (jklr) [reply telemetry/specs/INT.mdk]: I don't have a very strong preference but using 1 to indicate the need for conversion seems a bit more indicative and useful for debugging. Again, no strong opinion.

Comment (mickeyspiegel) [reply telemetry/specs/INT.mdk]: Done

File: telemetry/specs/INT.mdk (hunk 2 above)
Comment (mickeyspiegel) [reply telemetry/specs/INT.mdk]: Done

**Comment by mickeyspiegel** [comment]
Open Issues:

1. Should we reduce the INT-Type field to 4 bits as JK recently proposed?
2. Should we move the G bit to the Type byte, or leave it in the Reserved byte?

Note that the VXLAN GPE shim header format in draft-ietf-nvo3-vxlan-gpe-09 defines:
```
   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
   |      Type     |    Length     |    Reserved   | Next Protocol |
   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
```
where
Type:  This field MAY be used to identify different messages of this protocol.
Reserved:  The use of this field is reserved to the protocol defined in this message.

**Comment by jklr** [comment]
@mickeyspiegel as for the shim hdr format, I think the consensus from the last meeting is to follow the GPE convention, deviating from the INT convention.

**Review by jklr — APPROVED** [review]

**Pull request merged**